scripts/get_version.sh

```diff
@@ -28,4 +28,4 @@
 
 set -e
 
-grep -oP "PROJECT_VERSION \"?\K[0-9.]+(?=\")"? $(dirname "$0")/../CMakeLists.txt
+grep -oP "PROJECT_VERSION \"?\K[0-9.]+(?=\")?" $(dirname "$0")/../CMakeLists.txt
```

Comment:
The `?` character is meant to make the closing `"` in `PROJECT_VERSION "..."` optional and does - but only as long as the current directory does not contain a single-letter file name that would match it. In that case Bash would replace it with the file name and break the regex.

Putting it inside quotes fixes the problem.

Reply:
Are you sure this is correct? The `?` now seems to refer to the `(?`-expression preceding it, and not to a quote.

Reply:
Yeah. The quote is just Bash syntax and not a part of the regex. I think that we do want `?` to apply to the `(?`-expression (which is the part of the regex that will match a quote in the text). And it applies to it in both cases but if it's outside the quotes, it's interpreted as Bash wildcard syntax (unless nothing matches and then it falls back to being just `?`).